src/new/wallet.js

```diff
@@ -0,0 +1,439 @@
+  }
+
+  /**
+   * Connect to the server and start emitting events.
+   **/
+  start() {
+    const store = new MemoryStore();
+    storage.setStore(store);
+    storage.setItem('wallet:server', this.server);
+
+    ws.on('is_online', this.onConnectionChange);
+    ws.on('reload_data', this.reloadData);
+    ws.on('addresses_loaded', this.onAddressesLoaded);
+    ws.on('wallet', this.handleWebsocketMsg);
+
+    wallet.executeGenerateWallet(this.seed, this.passphrase, this.pinCode, this.password, false);
+
+    this.getTokenData();
+    this.serverInfo = null;
+    this.setState(HathorWallet.CONNECTING);
+
+    const promise = new Promise((resolve, reject) => {
+      version.checkApiVersion().then((info) => {
+        // Check network version to avoid blunders.
+        if (info.network.indexOf(this.network) >= 0) {
+          ws.setup();
+          this.serverInfo = info;
+          resolve(info);
+        } else {
+          this.setState(HathorWallet.CLOSED);
+          reject(`Wrong network. server=${info.network} expected=${this.network}`);
+        }
+      }, (error) => {
+        console.log('Version error:', error);
+        this.setState(HathorWallet.CLOSED);
+        reject(error);
+      });
+    });
+    return promise;
+  }
+
+  /**
+   * Close the connections and stop emitting events.
+   **/
+  stop() {
+    // TODO Double check that we are properly cleaning things up.
+    // See: https://github.com/HathorNetwork/hathor-wallet-headless/pull/1#discussion_r369859701
+    ws.stop()
+    ws.removeListener('is_online', this.onConnectionChange);
+    ws.removeListener('reload_data', this.reloadData);
+    ws.removeListener('addresses_loaded', this.onAddressesLoaded);
+    ws.removeListener('wallet', this.handleWebsocketMsg);
+    this.serverInfo = null;
+    this.setState(HathorWallet.CLOSED);
+  }
+
+  /**
+   * Returns the balance for each token in tx, if the input/output belongs to this wallet
+   */
+  getTxBalance(tx) {
+    const myKeys = []; // TODO
+    const balance = {};
+    for (const txout of tx.outputs) {
+      if (wallet.isAuthorityOutput(txout)) {
+        continue;
+      }
+      if (txout.decoded && txout.decoded.address
+          && txout.decoded.address in myKeys) {
+        if (!balance[txout.token]) {
+          balance[txout.token] = 0;
+        }
+        balance[txout.token] += txout.value;
+      }
+    }
+
+    for (const txin of tx.inputs) {
+      if (wallet.isAuthorityOutput(txin)) {
+        continue;
+      }
+      if (txin.decoded && txin.decoded.address
+          && txin.decoded.address in myKeys) {
+        if (!balance[txin.token]) {
+          balance[txin.token] = 0;
+        }
+        balance[txin.token] -= txin.value;
+      }
+    }
+
+    return balance;
+  }
+
+  getTokenData() {
+    if (this.tokenUid === HATHOR_TOKEN_CONFIG.uid) {
+      // Hathor token we don't get from the full node
+      this.token = HATHOR_TOKEN_CONFIG;
+    } else {
+      // Get token info from full node
```

Comment:
I'm not sure we need to do this. We don't seem to use the token name and symbol in the lib. If the user starts the wallet with a UID, but he probably already has this information. I think we should only have `this.tokenUid`.

Reply:
We do use the token name and symbol in the `prepareSendTokensData` method. We use it to return a pretty error message.

If we decide not to do this, we would require the user to pass the full token configuration when calling `sendTransaction`. If we refactor the `prepareSendTokensData` method, we could remove it but this would require another big refactor. I don't see why do this right now.

Reply:
Ok, let's refactor it later.